g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductComponentAction.java

```diff
@@ -180,5 +180,87 @@ public APIResponseDTO checkTranslationResult(String productName, String componen
 						en);
 		}
 		return m;
-	}	
+	}
+
+	/**
+	 * Get the the matched version from the version list by comparing to the input version
+	 *
+	 * @param productName
+	 * @param version
+	 * @return a matched version, if there's no matched version then return input version
+	 */
+	private String getMatchedVersion(final String productName, final String version) throws L3APIException{
+		Map<String, String[]> productsAndVersions = productService.getProductsAndVersions();
+		String matchedVersion = "";
+		if(productsAndVersions != null && !productsAndVersions.isEmpty()) {
+			String[] versionList = productsAndVersions.get(productName);
+			if(versionList != null && versionList.length > 0) {
+				for(String s : versionList) {
+					String f = filterVersion(s, version);
+					if(compare(f, version) == -1 && compare(s, matchedVersion) == 1) {
```

Comment:
Do s and machedVersion have same length? Is is necessary to filterVersion for machedVersion and compare it with s?

Reply:
good catch!